tests/e2e/performance/test_small_file_workload.py

```diff
@@ -323,9 +317,130 @@ class TestSmallFileWorkload(PASTest):
+
+        # Calculating the size of the volume that need to be test, it should
+        # be at least twice in the size then the size of the files, and at
+        # least 100Gi.
+        # Since the file_size is in Kb and the vol_size need to be in Gb, more
+        # calculation is needed.
+        vol_size = int(files * threads * file_size * 3)
+        vol_size = int(vol_size / constants.GB2KB)
+        if vol_size < 100:
+            vol_size = 100
+        self.crd_data["spec"]["workload"]["args"]["storagesize"] = f"{vol_size}Gi"
+
+    def init_full_results(self, full_results):
+        """
+        Initialize the full results object which will send to the ES server
+
+        Args:
+            full_results (obj): an empty SmallFileResultsAnalyse object
+
+        Returns:
+            SmallFileResultsAnalyse (obj): the input object fill with data
+
+        """
+        for key in self.environment:
+            full_results.add_key(key, self.environment[key])
+
+        # Calculating the total size of the working data set - in GB
+        full_results.add_key(
+            "dataset",
+            self.crd_data["spec"]["workload"]["args"]["file_size"]
+            * self.crd_data["spec"]["workload"]["args"]["files"]
+            * self.crd_data["spec"]["workload"]["args"]["threads"]
+            * full_results.results["clients"]
+            / constants.GB2KB,
+        )
+
+        full_results.add_key(
+            "global_options",
+            {
+                "files": self.crd_data["spec"]["workload"]["args"]["files"],
+                "file_size": self.crd_data["spec"]["workload"]["args"]["file_size"],
+                "storageclass": self.crd_data["spec"]["workload"]["args"][
+                    "storageclass"
+                ],
+                "vol_size": self.crd_data["spec"]["workload"]["args"]["storagesize"],
+            },
+        )
+        return full_results
+
+    def run(self):
+        log.info("Running SmallFile bench")
+        self.deploy_and_wait_for_wl_to_start(timeout=240, sleep=10)
+
+        # Getting the UUID from inside the benchmark pod
+        self.uuid = self.ripsaw.get_uuid(self.client_pod)
+        self.wait_for_wl_to_finish(sleep=30)
+        try:
+            if "RUN STATUS DONE" in self.test_logs:
+                log.info("SmallFiles has completed successfully")
+                return True
+        except IOError:
+            log.warning("SmallFiles failed to complete")
+            return False
+
+    def teardown(self):
+        """
+        The teardown of the test environment in the end.
+
+        """
+        log.info("cleanup the environment")
+        if hasattr(self, "ripsaw"):
+            self.ripsaw.cleanup()
+        if isinstance(self.es, ElasticSearch):
+            self.es.cleanup()
+
+        sleep_time = 5
+        log.info(
+            f"Going to sleep for {sleep_time} Minute, for background cleanup to complete"
+        )
+        time.sleep(sleep_time * 60)
```

Comment:
same as above - simply sleep for 300 seconds.

Reply:
same as above.